src/client/camera.cpp

```diff
@@ -557,6 +587,47 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 tool_reload_ratio)
 
 	m_wieldnode->setNodeLightColor(player->light_color);
 
+	// Offhand wield Item
+	v3f offhand_wield_position = v3f(m_wieldmesh_offset.X - 110, m_wieldmesh_offset.Y, 65);
+	v3f offhand_wield_rotation = v3f(-100, 120, -100);
+	offhand_wield_position.Y += fabs(m_offhand_wield_change_timer)*320 - 40;
+	if(m_offhand_digging_anim < 0.05 || m_offhand_digging_anim > 0.5)
+	{
+		f32 frac = 1.0;
+		if(m_offhand_digging_anim > 0.5)
+			frac = 2.0 * (m_offhand_digging_anim - 0.5);
+		f32 ratiothing = std::pow((1.0f - tool_reload_ratio), 0.5f);
+		f32 ratiothing2 = (easeCurve(ratiothing*0.5))*2.0;
+		offhand_wield_position.Y -= frac * 25.0 * pow(ratiothing2, 1.7f);
+		offhand_wield_position.X += frac * 35.0 * pow(ratiothing2, 1.1f);
+		offhand_wield_rotation.Y += frac * 70.0 * pow(ratiothing2, 1.4f);
+	}
+	if (m_offhand_digging_button != -1)
+	{
+		f32 digfrac = m_offhand_digging_anim;
+		offhand_wield_position.X += 50 * sin(pow(digfrac, 0.8f) * M_PI);
+		offhand_wield_position.Y += 24 * sin(digfrac * 1.8 * M_PI);
+		offhand_wield_position.Z -= 25 * 0.5;
+
+		core::quaternion quat_begin(offhand_wield_rotation * core::DEGTORAD);
+		core::quaternion quat_end(v3f(-80, 30, -100) * core::DEGTORAD);
+		core::quaternion quat_slerp;
+		quat_slerp.slerp(quat_begin, quat_end, sin(digfrac * M_PI));
+		quat_slerp.toEuler(offhand_wield_rotation);
+		offhand_wield_rotation *= core::RADTODEG;
+	} else {
+		f32 bobfrac = my_modf(m_view_bobbing_anim);
+		offhand_wield_position.X += sin(bobfrac*M_PI*2.0+M_PI) * 3.0;
+		offhand_wield_position.Y += sin(my_modf(bobfrac*2.0)*M_PI+M_PI) * 3.0;
+	}
+	m_offhand_wieldnode->setPosition(offhand_wield_position);
+	m_offhand_wieldnode->setRotation(offhand_wield_rotation);
```

Comment:
Better proposal: Move the needed values to a struct and have one instance for left and one for right. It could be used for shared function calls to reduce code bloat.